### DISCUSSION TITLE
Killing the web socket connection when modal closes

### DISCUSSION THREAD

**Discussion by tesla-srt** [k-yle/rtsp-relay#42] [Q&A] [answered]
Hi, I have an app that I am working on that can handle multiple cameras.
each camera has a bootstrap modal that displays the canvas.

I am wondering how I can ensure that the web socket connection the rtsp stream is killed when the modal gets closed

**Comment by k-yle** [comment, marked as answer]
Hi, if you initialized the player on the client side using `loadPlayer` (see recently updated README), then you make use of the [`Player`](https://github.com/k-yle/rtsp-relay/blob/main/browser/jsmpeg.d.ts#L59-L85) object that is returned by that promise.

For example:
```js
const player = await loadPlayer({
  url: 'ws://' + location.host + '/api/stream',
  canvas: document.getElementById('canvas')
});

player.destroy();
```

If you want to be certain that it works, on the server side, add `verbose: true` to `proxy({ url: "..." })`and you'll see a message printed that says `"no clients left; destroying stream"`

Hope this helps!

**Reply by tesla-srt** [reply]
Aha! Thank you!

**Reply by rakeshraushan1997** [reply]
but this is not working raise error  "Module parse failed: Cannot use keyword 'await' outside an async function (17:23)
File was processed with these loaders:"

**Reply by k-yle** [reply]
@rakeshraushan1997 you may want to [read this introduction to async functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function)

**Comment by tesla-srt** [comment]
@k-yle I noticed that .destroy() also deletes the canvas?

**Reply by k-yle** [reply]
I think it might do that... not at my laptop rn so I can't confirm. 

Perhaps `.stop()` will work better for you?
